File: src/problem23.hs
import qualified Data.Set as Set

isAbundant :: Int -> Bool
isAbundant n = n < (sum $ filter (\x -> n `mod` x == 0) [1..(n `div` 2)])

answer :: Int
answer = sum $ (Set.fromList [1..28123]) Set.\\ abundantsPlus 
    where
        abundants = filter isAbundant [12..28123]
        abundantsPlus = Set.fromList [x+y | x<-abundants,y<-abundants]

File: src/problem23.py
def isAbundant(n):
    total = 0
    for i in range(1,(n//2)+1):
        if n % i == 0:
            total += i
    if total > n:
        return True
    return False

def problem23():
    abundants = []
    for i in range(12,28124):
        if isAbundant(i):
            abundants.append(i)
    abundantSums = [i+j for i in abundants for j in abundants]
    values = range(1,28124)
    return sum(list(set(values) - set(abundantSums)))

print(problem23())
